commands/reminder.js

```diff
@@ -0,0 +1,143 @@
+        .addChoices(
+            {name: 'Daily',value: 'daily'},
+            {name: 'Weekly', value:'weekly'},
+            {name: 'Monthly', value: 'monthly'},
+            {name:'Bi-weekly (once every two weeks)', value:'biweekly'},
+            {name:'None', value:'none'}
+        )
+    ),
+
+    async execute(interaction) {
+        await db; // Wait for the database connection
+
+        const channel = interaction.options.getChannel('channel');
+        const date = chrono.parseDate(interaction.options.getString('date'));
+        const message = interaction.options.getString('message');
+        const role = interaction.options.getRole('role');
+        const recurrence = interaction.options.getString('recurrence');
+
+        try {
+            let newReminder;
+
+            newReminder = new scheduledScrimSchema({
+                date: new Date(date),
+                content: message,
+                channel: channel.id,
+                role: role.id,
+                recurrence: recurrence,
+            });
+
+            await newReminder.save();
+
+            // Schedule the reminder for this scrim
+            await scheduleReminder(newReminder, recurrence);
+
+            // Send a confirmation DM to the user who wrote the command
+            interaction.user.send(`You have a reminder scheduled: "${message}" in ${channel} on ${date}`);
+            // The little 'only you can see this' thing that the bot will send
+            return interaction.reply({ content: `Scheduled "${message}" in ${channel} on ${date}`, ephemeral: true });
+
+        } catch (error) {
+            console.error('Error saving reminder:', error);
+            return interaction.reply({ content: 'An error occurred while scheduling the reminder.', ephemeral: true });
+        }
+
+    },
+};
+
+// Recursive function to handle scheduling for recurring reminders
+function scheduleRecurringJob(reminder, recurrence) {
+    schedule.scheduleJob(reminder.date, async () => {
+        sendChannelReminder(reminder);
+        updateDate(reminder, recurrence);
+        await reminder.save();
+        scheduleRecurringJob(reminder, recurrence);
+    });
+}
+
+function updateDate(reminder, recurrence) {
+    // Update the date based on the recurrence type
+    switch (recurrence) {
+        case 'daily':
+            reminder.date.setDate(reminder.date.getDate() + 1);
+            break;
+        case 'weekly':
+            reminder.date.setDate(reminder.date.getDate() + 7);
+            break;
+        case 'monthly':
+            reminder.date.setMonth(reminder.date.getMonth() + 1);
+            break;
+        case 'biweekly':
+            reminder.date.setDate(reminder.date.getDate() + 14);
+            break;
+        default:
+            break;
+    }
+}
+
+// Method to handle scheduling logic for reminders
+async function scheduleReminder(reminder, recurrence) {
```

Comment:
Consider creating multiple files, one that might add the reminders and another that would handle the sending of them. Perhaps the one that saves it would only save it to the db, and the one that sends it could then fetch from the db. They would still be linked, but easier to follow.